cfgmgr/natmgrd.cpp

```diff
@@ -98,6 +102,12 @@ void sigterm_handler(int signo)
         natmgr->cleanupMangleIpTables();
         natmgr->cleanupPoolIpTable();
     }
+
+#if defined(ASAN_ENABLED)
+    __lsan_do_leak_check();
+#endif
+    signal(signo, SIG_DFL);
+    raise(signo);
```

Comment:
Do you intend to move code into above #if block?

Reply:
@qiluo-msft  yes, there is an issue here with closing an application after clean-up. 
Supervisord sends SIGTERM to all applications that it controls. Application after receiving the signal may optionally run a clean-up routine and should exit. natmngrd runs a cleanup from the signal handler but doesn't exit after that. After supervisord sends a signal it waits for 10 seconds (default timeout) to allow the application to exit correctly. If the application is not exiting in the given timeout supervisord kills the application. natmngrd is always killed by the supervisord. This affects swss container restart time. To exit from the user-defined signal_handler application must restore the default signal handler and raise the signal to itself again.
In the latest commits to this PR I removed this change to not mix it with ASAN changes and will create a separated PR for a bug fix.

Reply:
Thanks for reporting this bug! Could you separate the bug fix into a standalone PR to make backport easier?

Suggest follow the practice of using `sig_atomic_t` and gracefully handle it in the main loop. This is not a trivial fix but worthy.

Reply:
https://github.com/Azure/sonic-swss/pull/2232